src/thrift-client/TgtInterfaceImpl.cpp

```diff
@@ -1740,3 +1750,31 @@ int HycGetVmdkStats(const char* vmdkid, vmdk_stats_t *vmdk_stats)
 	return 0;
 }
 
+int HycGetComponentStats(component_stats_t *g_stats)
+{
+	std::vector<::hyc::StordVmdk*> vmdks = g_stord.GetVmdkList();
```

Comment:
Please consider race between delete VMDK and stats.

Reply:
thanks. have put in a fix. please have a look.